package.json

```diff
@@ -198,12 +198,14 @@
           "type": "string",
           "enum": [
             "Paginate",
-            "Infinite"
+            "Infinite",
+            "Auto-Append"
           ],
           "default": "Paginate",
           "enumDescriptions": [
             "Maintains a consistent memory size, replacing the previous request.",
-            "Appends new memory to bounds of current request, resulting in a growing list."
+            "Appends new memory to bounds of current request, resulting in a growing list.",
+            "Appends new memory to bounds of current request if the bottom of the list is reached, resulting in an automatically growing list."
```

Comment:
The current implementation of `Infinite` requires the user to click on `show x more bytes below` manually to append new memory. While the `Auto-Append` implementation requires the user only to scroll to the bottom of the list. Then in the background new memory will be fetched automatically and appended.

So, in reality, `Infinite` is not really *automatic*.

Reply:
Hm... that doesn't sound like infinite scroll to me, just scrolling, and we may not need all three options. It appears that the only reference to the `Infinite` variant is in this `loadMoreMemory` method:

```
    const loadMoreMemory = (e: React.MouseEvent | React.KeyboardEvent): void => {
        containerRef.current?.blur();
        if (isTrigger(e)) {
            let newOffset = memoryReadArguments.offset;
            let newCount = memoryReadArguments.count;
            if (direction === 'above') {
                newOffset = memoryReadArguments.offset - numBytes;
            }
            if (scrollingBehavior === 'Infinite') {
                newCount = memoryReadArguments.count + numBytes;
            } else {
                if (direction === 'below') {
                    newOffset = memoryReadArguments.offset + numBytes;
                }
            }
            fetchMemory({ offset: newOffset, count: newCount });
        }
    };
```

I think the behavior there is actually wrong in any case. In the `else` block, we're not loading more memory, we're loading the same amount, but from a different starting point. As a user, that wouldn't match my expectations of what loading more memory would mean. I suggest that we simply remove the `Infinite` variant, and in this method increment the `count` unconditionally, modifying the `offset` only if memory is added above the current block.

Reply:
Basically, the current code is mixing two notions:
 - What triggers a change in the memory loaded. (Infinite scroll enabled/disabled)
 - How much memory is loaded when a change occurs. (Paginate vs. grow)

For the first question, our current answers are "on button click" or "on scroll", and for the second question, our answers are "same amount" or "more". We could separate those into separate preferences for clarity, or we should be more explicit in all user-facing text (preference descriptions and UI text of the more memory header/footer) about the combination of behaviors that will be enacted.

Reply:
I think the `Infinite` variant was confusing from the start. If there are no strong advocates for that one, then I'd agree to remove it or repurpose it.

We initially had a discussion about splitting into two options. But weren't fully convinced of the "on scroll" + "same amount" combination. Plus the complexity for the user to fiddle around with two options while there are max. four combinations (of which only three looked useful). I'd be in favor of keeping a single setting with clearer messaging to users which in itself would be helpful to them.

Reply:
> I think the behavior there is actually wrong in any case. In the else block, we're not loading more memory, we're loading the same amount, but from a different starting point. 

The `loadMoreMemory` is called after clicking on the buttons `Load x more bytes above/below` and consequently also used by the `Paginate` scroll behavior. The reason for the `else` block was probably to allow `Paginate` to work properly.

> and in this method increment the count unconditionally, modifying the offset only if memory is added above the current block.

This would also change how `Paginate` works.

I would suggest to rename the method to `updateMemory` and trigger different methods based on the active scrollbehavior, e.g., `createPaginateRequest(...)`, `createXRequest(...)`